builtin/v10/migration/miner.go

```diff
@@ -0,0 +1,437 @@
+			}
+
+			outDeadlines.Due[i] = outDlCid
+		}
+	}
+
+	return store.Put(ctx, &outDeadlines)
+}
+
+func (m minerMigrator) migratePartitions(ctx context.Context, store adt9.Store, cache migration.MigrationCache, partitions cid.Cid) (cid.Cid, error) {
+	return cache.Load(migration.PartitionsAmtKey(partitions), func() (cid.Cid, error) {
+		inPartitionsArr, err := adt9.AsArray(store, partitions, miner9.DeadlinePartitionsAmtBitwidth)
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to load partitions: %w", err)
+		}
+
+		outPartitions, err := adt10.MakeEmptyArray(store, miner10.DeadlinePartitionsAmtBitwidth)
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to create new partitions: %w", err)
+		}
+
+		var inPartition miner9.Partition
+		err = inPartitionsArr.ForEach(&inPartition, func(i int64) error {
+			outEsCid, err := m.migrateExpirations(ctx, store, cache, inPartition)
+			if err != nil {
+				return xerrors.Errorf("failed to migrate expirations: %w", err)
+			}
+
+			outPartition := miner10.Partition{
+				Sectors:           inPartition.Sectors,
+				Unproven:          inPartition.Unproven,
+				Faults:            inPartition.Faults,
+				Recoveries:        inPartition.Recoveries,
+				Terminated:        inPartition.Terminated,
+				ExpirationsEpochs: outEsCid,
+				EarlyTerminated:   inPartition.EarlyTerminated,
+				LivePower:         miner10.PowerPair(inPartition.LivePower),
+				UnprovenPower:     miner10.PowerPair(inPartition.UnprovenPower),
+				FaultyPower:       miner10.PowerPair(inPartition.FaultyPower),
+				RecoveringPower:   miner10.PowerPair(inPartition.RecoveringPower),
+			}
+
+			err = outPartitions.Set(uint64(i), &outPartition)
+			if err != nil {
+				return xerrors.Errorf("failed to set partition: %w", err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to iterate over partitions: %w", err)
+		}
+
+		return outPartitions.Root()
+	})
+}
+
+func (m minerMigrator) migrateExpirations(ctx context.Context, store adt9.Store, cache migration.MigrationCache, inPartition miner9.Partition) (cid.Cid, error) {
+	return cache.Load(migration.ExpirationsAmtKey(inPartition.ExpirationsEpochs), func() (cid.Cid, error) {
+		// We aren't going to use the quant spec
+		expirationQueue, err := miner9.LoadExpirationQueue(store, inPartition.ExpirationsEpochs, builtin.QuantSpec{}, miner9.PartitionExpirationAmtBitwidth)
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("loading expiration queue: %w", err)
+		}
+
+		outExpirations, err := adt10.MakeEmptyArray(store, miner10.DeadlineExpirationAmtBitwidth)
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to create new expiration queue: %w", err)
+		}
+
+		var inExpirationSet miner9.ExpirationSet
+		err = expirationQueue.ForEach(&inExpirationSet, func(i int64) error {
+			newExpirationSet := miner10.ExpirationSet{
+				OnTimeSectors: inExpirationSet.OnTimeSectors,
+				EarlySectors:  bitfield.New(),
```

Comment:
@Kubuxu can probably explain this one better, but we don't set any ProofExpiringSectors in this migration because we set the proof expiration to be `at the same time or after` the commitment expiration. This means that we will definitely have to check this sector in the faulty queue before the proof would expire. Obviously if the sector commitment is extended, we might have to make an entry in the proof expiration queue.